channels/README.md

```diff
@@ -0,0 +1,368 @@
+|   |                | initialized | ------------------------------------> |   |
+|   | <------------- +-------------+                                       |   |
+|   |   [timeout]/          | channel_accept                               |   |
+|   |   [disconnect]        v                                              |   |
+|   |                +-------------+                 error                 |   |
+|   | <------------- |  accepted   | ------------------------------------> |   |
+| c |   [timeout]/   +-------------+                                       | e |
+| l |   [disconnect]        | funding_created                              | r |
+| o |                       v                                              | r |
+| s |                +-------------+                 error                 | o |
+| e | <------------- | half_signed | ------------------------------------> | r |
+| d |   [timeout]/   +-------------+                                       |   |
+|   |   [disconnect]        | funding_signed                               |   |
+|   |                       v                                              |   |
+|   |                +-------------+  [disconnect]                         |   |
+|   | <------------- |   signed    | -------------+  error                 |   |
+|   |   [timeout]    +-------------+ -------------|----------------------> |   |
+|   |                       |    |  shutdown      |                        |   |
+|   |        funding_locked |    +------------+   |                        |   |
+|   |                       |             |   |                        |   |
+|   |                       |                 |   v                        |   |
+|   |                       v        [disconnect] +--------------+  error  |   |
+|   |                +-------------+ ---------|-> | disconnected | ------> |   |
+|   | <------------- |    open     |          |   +--------------+         |   |
+|   |   [timeout]    +-------------+ <--------|------------+               |   |
+|   |                 ^ |      |       channel_reestablish                 |   |
+|   |       update_*/ | |      |              |                            |   |
+|   |       ping/pong +-+      | shutdown     v                            |   |
+|   |                          |    +-------------+ [disconnect]           |   |
+|   |                          +--> |   closing   | ------------+          |   |
+|   |                               +-------------+             |          |   |
+|   |                                |  ^                       |          |   |
+|   |                                |  | channel_reestablish   |          |   |
+|   | <------------------------------+  |                       v          |   |
+|   |       closing_signed              |         +--------------+  error  |   |
+|   |                                   +-------- | disconnected | ------> |   |
```

Comment:
There are two states `disconnected`, with event `channel_reestablish` leading to distinct states. So maybe the `disconnected` states shall be renamed e.g. disconnected1 and disconnected2 ?